### ISSUE TITLE
Provide better error message if custom theme files can not be parsed

### ISSUE THREAD

**Issue by dustypomerleau** [sharkdp/bat#1543]
**What version of `bat` are you using?**

0.17.1

**Describe the bug you encountered:**

Placing new `.tmTheme` files in `~/.config/bat/themes` has no effect when running `bat cache --build`.
Output: `No themes were found in '/Users/me/.config/bat/themes', using the default set`
Running the environment script below states that custom themes are found, but running `bat --list-themes` doesn't show the new themes.

**What did you expect to happen instead?**

New `.tmTheme` files should be added to the cache.

**How did you install `bat`?**

`brew`

---

**Environment**
system
------

**$ uname -srm**
Darwin 20.3.0 x86_64  

**$ sw_vers**
ProductName:    macOS  
ProductVersion: 11.2.1  
BuildVersion:   20D74  

bat
---

**$ bat --version**
bat 0.17.1  

**$ env**
PAGER=less  
BAT_THEME=Test  

**$ bat --list-languages**
Found custom syntax set.

**$ bat --list-themes**
Found custom theme set.

bat_config
----------

bat_wrapper
-----------

No wrapper script for 'bat'.

bat_wrapper_function
--------------------

No wrapper function for 'bat'.

**$ /usr/local/bin/zsh --login -i -c type cat**
```
cat is an alias for colorize_cat
```

tool
----

**$ less --version**
less 487 (POSIX regular expressions)

**Comment by sharkdp** [comment]
Could you please show us the `.tmTheme` file? Maybe attach it here or upload to https://gist.github.com/

Also, where exactly did you place it beneath the `.config/bat/themes` folder?

**Comment by dustypomerleau** [comment]
I haven't solved this problem, but I'm going to close, because additional tinkering is making it clear this issue is related to my `.tmTheme` syntax. I am trying to convert 2 of my VS Code themes for use with Bat, and they are in JSON ([Yarra Valley](https://github.com/dustypomerleau/yarra-valley/blob/master/themes/yarra-valley.json) and [Tol](https://github.com/dustypomerleau/tol/blob/master/themes/tol.json)). After some manual editing to remove a lot of VS Code-specific color assignments, I was able to get one of the themes into the cache, but it isn't yet highlighting the tokens. I think this is because my method of conversion is creating a separate `<string>` tag for each array member inside `<scope>`, whereas the themes in this repo appear to have a single `<string>` tag for the entire array of scopes.

You can see what the tweaked file conversion looks like in [this gist](https://gist.github.com/dustypomerleau/d58f71bb27e9ddbdf8b013fee88637ce). I think I will need to edit out all of the extra `<string>` tags somehow.

If you know of any way to easily/directly convert the JSON to `tmTheme`, I'd be very interested to try it. Most tooling appears designed to go the other direction, unfortunately.

In any event, thanks for all your hard work on Bat. When I manage to get these converted, I'll submit a PR if you're receptive to adding more themes.

**Comment by sharkdp** [comment]
I'd like to reopen it, because I think we could have definitely provided you with a better error message.

**Comment by sharkdp** [comment]
> If you know of any way to easily/directly convert the JSON to `tmTheme`, I'd be very interested to try it. Most tooling appears designed to go the other direction, unfortunately.

Maybe @keith-hall would know?

**Comment by keith-hall** [comment]
Unfortunately I can't recommend anything - I tried the tool I would have recommended (the PackageDev package in Sublime Text) and it suffered from the same problem. I guess the array of strings VSCode uses for `scope` in it's JSON is a little unusual, in that tools that convert to tmTheme/PLIST format do a literal conversion, which is why we end up with `<key>scope</key><array><string>scope1</string><string>scope2</string>` etc. instead of `<key>scope</key><string>scope1, scope2</string>`. Interestingly, trying the array format in Sublime Text just crashed it, so I don't think it's something we'd necessarily want to update syntect to handle, although it would make it more convenient to use these converted color schemes.

**Comment by sharkdp** [comment]
It seems like `bat` should have printed an error along the lines of
```
Failed to load one or more themes from '/home/shark/.config/bat/themes' (reason: '…')
```

If it didn't, please let me know and share the file that was not converted properly.

**Comment by sharkdp** [comment]
Ahh.. I think this was only just implemented in #1453. Should be fixed in v0.18.
